translate/storage/pypo.py

```diff
@@ -757,20 +760,23 @@ class pofile(pocommon.pofile):
     """A .po file containing various units"""
     UnitClass = pounit
 
-    def parse(self, input):
+    def parse(self, input, duplicatestyle=None):
         """Parses the given file or file source string."""
-        if True:
-            if hasattr(input, 'name'):
-                self.filename = input.name
-            elif not getattr(self, 'filename', ''):
-                self.filename = ''
-            if isinstance(input, str):
-                input = BytesIO(input)
-            # clear units to get rid of automatically generated headers before parsing
-            self.units = []
-            poparser.parse_units(poparser.ParseState(input, pounit), self)
-
-    def removeduplicates(self, duplicatestyle="merge"):
+        if hasattr(input, 'name'):
+            self.filename = input.name
+        elif not getattr(self, 'filename', ''):
+            self.filename = ''
+        if isinstance(input, str):
+            input = BytesIO(input)
+        # clear units to get rid of automatically generated headers before parsing
+        self.units = []
+        poparser.parse_units(poparser.ParseState(input, pounit), self)
+
+        # duplicates are now removed by default unless duplicatestyle=allow
+        if duplicatestyle != "allow":
```

Comment:
Just noticed what a stupid thing is to have `def parse(self, input, duplicatestyle="allow"):` since having mutable objects as default values is highly discouraged. So it should be instead:

```python
        if duplicatestyle is not None and duplicatestyle != "allow":
```


@phlax What I was expecting of this PR is to keep previous behavior by default, but allowing to discard duplicates in case it was explicitly specified when calling `parse()` (necessary for Pootle). I was hoping that this would be a non-invasive change that didn't require to touch existing tests at all, despite it would be necessary to add new tests to test new behavior. Have you tried that approach?